images/hub/Dockerfile

```diff
@@ -21,9 +21,11 @@ FROM python:3.11-bullseye as build-stage
 COPY requirements.txt requirements.txt
 ARG PIP_CACHE_DIR=/tmp/pip-cache
 RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
-    pip install build \
```

Comment:
I'm not confident why we installed `build` here, so I'm not confident on seeing it removed - can you motivate the change?

Reply:
It was added but never used without comment in [#2733](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2733/files#diff-b55155cca69fb062e0d3eb1511e7754d1099bedf339294ea738cfd925a490753R14).

I believe it was added by mistake, copying what we do in binderhub, which includes building the binderhub wheel itself, which used to be done with `python3 -m build --wheel .` (not anymore). If you're not building _local_ packages from source via `python3 -m build` (and not `pip wheel`, which can also do the same thing), `build` is never used.

From what I can find, it's here because we used to [invoke it here in binderhub](https://github.com/jupyterhub/binderhub/commit/9bf1418c62c20a10d565e6e3996e7b24edb98f8a#diff-d91e821ae7349a5e3cce784da75cb2695c349a4096193500bede2da5010987e3L28), but since  https://github.com/jupyterhub/binderhub/pull/1535/files switched to invoking `pip wheel`, `build` has not been used in either binderhub or z2jh. It's a tiny unused pure-python package in the build stage, so nobody noticed the very tiny bit of wasted time.

Reply:
Ah, thank you @minrk!